src/caches/abstractbeliefpropagationcache.jl

```diff
@@ -24,26 +25,48 @@ function data_graph_type(bpc::AbstractBeliefPropagationCache)
 end
 data_graph(bpc::AbstractBeliefPropagationCache) = data_graph(tensornetwork(bpc))
 
-function default_message_update(contract_list::Vector{ITensor}; normalize=true, kwargs...)
-  sequence = contraction_sequence(contract_list; alg="optimal")
-  updated_messages = contract(contract_list; sequence, kwargs...)
+function message_update_function(
+  alg::Algorithm"contract",
+  contract_list::Vector{ITensor};
+  normalize=alg.kwargs.normalize,
+  sequence_alg=alg.kwargs.sequence_alg,
```

Comment:
I would lean towards not having these keyword arguments and just have the keyword arguments passed through `alg`, I think otherwise it starts getting confusing keeping track of how to pass the keyword arguments, which ones take precedence, etc.